File: go/v1/client/environment.go
package client

// Environment is the target environment for the PayGate API.
type Environment struct {
	baseURL  string
	basePath string
}

var (
	// Sandbox testing environment.
	SANDBOX = Environment{"https://backend.hunanxiaojunzi.com", "/api/sandbox/gateway/v1"}
	// Live production environment.
	LIVE = Environment{"https://backend.hunanxiaojunzi.com", "/api/gateway/v1"}
)

// BaseURL returns the full API base URL.
func (e Environment) BaseURL() string { return e.baseURL + e.basePath }

// BasePath returns the path prefix for HMAC signing.
func (e Environment) BasePath() string { return e.basePath }

File: php/v1/src/Environment.php
<?php

namespace Paygate\Sdk;

/**
 * Target environment for the PayGate API.
 *
 * @author alphah
 * @since 1.0.0
 */
class Environment
{
    /** Sandbox testing environment. */
    const SANDBOX = ['host' => 'https://sandbox.backend.hunanxiaojunzi.com', 'basePath' => '/api/sandbox/gateway/v1'];

    /** Live production environment. */
    const LIVE = ['host' => 'https://api.backend.hunanxiaojunzi.com', 'basePath' => '/api/gateway/v1'];

    public static function baseUrl(array $env): string
    {
        return $env['host'] . $env['basePath'];
    }

    public static function basePath(array $env): string
    {
        return $env['basePath'];
    }
}
